cmd/config-versions.go

```diff
@@ -564,3 +565,23 @@ type serverConfigV22 struct {
 	// Notification queue configuration.
 	Notify *notifier `json:"notify"`
 }
+
+// serverConfigV23 is just like version '22' with addition of cache field
+type serverConfigV23 struct {
+	sync.RWMutex
+	Version string `json:"version"`
+
+	// S3 API configuration.
+	Credential auth.Credentials `json:"credential"`
+	Region     string           `json:"region"`
+	Browser    BrowserFlag      `json:"browser"`
+	Domain     string           `json:"domain"`
+
+	// Storage class configuration
+	StorageClass storageClassConfig `json:"storageclass"`
+
+	// Cache configuration
+	Cache *cacheConfig `json:"cache"`
```

Comment:
Correct way to define cache config is:
```
type serverConfigV23 struct {
	sync.RWMutex
	Version string `json:"version"`

	// S3 API configuration.
	Credential auth.Credentials `json:"credential"`
	Region     string           `json:"region"`
	Browser    BrowserFlag      `json:"browser"`
	Domain     string           `json:"domain"`

	// Storage class configuration
	StorageClass storageClassConfig `json:"storageclass"`

	// Cache configuration
	Cache struct {
		Drives  []string `json:"drives"`
		Expiry  int      `json:"expiry"`
		Exclude []string `json:"exclude"`
	} `json:"cache"`

	// Notification queue configuration.
	Notify *notifier `json:"notify"`
}
```

Because for arguments sake let's say in future we decide to change `Expiry` to string type which takes `"30d"` `"12h"`, if we do that json.Unmarshal() will fail during migration. Another way to fix it is do `type CacheConfigV1 struct {}` and include it in the serverConfig23. It is easier to "inline" the struct (as shown above) instead of defining a new struct type.

This argument applies to notifier too. If we change the type of any of the fields then migration will break.